### DISCUSSION TITLE
Why use FIFO IO directly

### DISCUSSION THREAD

**Discussion by jokemanfire** [containerd/rust-extensions#238] [Q&A]
I find this issus  from 
[https://github.com/containerd/containerd/issues/9568](url)

The community shim links child procress through copyIO. The shim and containerd have separate pipe links, so the shim can close the pipe when the exec process exit. r-shim use the FIFO-IO directly.

**Comment by jokemanfire** [comment]
@abel-von @mxpv  @Burning1020
problem  find in ctr v1.7 "TestContainerDrainExecIOAfterExit"

**Comment by Burning1020** [comment]
As mentioned here: https://github.com/containerd/rust-extensions/pull/24#issuecomment-1040619960
We think directly IO could be efficient and simple.

**Comment by jokemanfire** [comment]
but the exec process will  occupy the fifo io , so the containerd in execsync function will loop forver ,in test "TestContainerDrainExecIOAfterExit " .how can we resolve this problem . that's why I pull this pr https://github.com/containerd/rust-extensions/pull/235  . How to report errors at the underlying runtime , the runc can't trace the daemon like "sleep 365d &" , and 
 runc didn't return the error .  I hope we can resolve this problem , if it is runc's problem .do you have any suggestions?
